python/graphstorm/model/gnn_distill/student.py

```diff
@@ -0,0 +1,127 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GNN student model class
+"""
+import os
+import torch as th
+from torch import nn
+from transformers import (
+    AutoTokenizer,
+    AutoConfig,
+    AutoModel,
+    DistilBertConfig, 
+    DistilBertModel, 
+)
+
+SUPPORTED_MODEL = {
+"DistilBertModel": (DistilBertConfig, DistilBertModel),
+}
+
+PRETRAINED_MODEL = [
+"distilbert-base-uncased",
+]
+
+class GSDistilledModel(nn.Module):
+    """ GraphStorm GNN distilled model class.
+    User specified Transformer-based model 
+    and a projection matrix will be initialized.
+
+    Parameters
+    ----------
+    lm_name : str
+        Model name for Transformer-based student model.
+    gnn_embed_dim : int
+        Dimension of GNN embeddings.
+    pre_trained_name : str
+        Name of pre-trained model.
+    """
+    def __init__(self, lm_name=None, pre_trained_name=None, checkpoint_path=None):
+        super(GSDistilledModel, self).__init__()
+
+        if not checkpoint_path:
+            # TODO (HZ): need to test other HF models.
+            if lm_name not in SUPPORTED_MODEL:
+                raise ValueError(f'Model class {lm_name} is not supported.')
+            if pre_trained_name is not None and pre_trained_name not in PRETRAINED_MODEL:
+                raise ValueError(f'Pre-trained model {pre_trained_name} is not supported.')
+
+            # initiate Transformer-based model
+            self.lm_config = SUPPORTED_MODEL[lm_name][0]()
+            self.lm = SUPPORTED_MODEL[lm_name][1](self.lm_config)
+
+            # load pre-trained parameters if any
+            if pre_trained_name is not None:
+                self.tokenizer = AutoTokenizer.from_pretrained(pre_trained_name)
+                self.lm = self.lm.from_pretrained(pre_trained_name)
+            else:
+                # default tokenizer
+                self.tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased")
+        else:
+            self.load_from_gs_checkpoint(checkpoint_path)
+
+        # TODO (HZ): support more distance-based loss 
+        self.loss = nn.MSELoss()
+
+    def load_from_gs_checkpoint(self, checkpoint_path):
```

Comment:
i try to understand this functionality. the checkpoint saves the projection parameters, but it doesn't save the transformer model parameters? when this checkpointing should be used?

Reply:
it saves both as well as tokenizer. This load_from_gs_checkpoint will be used when loading the distilled student model in student inference implementation, which is not in this PR.